services/ui-src/src/components/fields/CheckboxField.tsx

```diff
@@ -15,32 +13,6 @@ export const CheckboxField = ({
   ...props
 }: ChoiceFieldProps) => {
   const mqClasses = makeMediaQueryClasses();
-  const [fieldValues, setFieldValues] = useState<string[] | null>(null);
-
-  // get the form context
-  const form = useFormContext();
-
-  // update local state
-  const onChangeHandler = (event: InputChangeEvent) => {
-    const checked = event.target.checked;
-    const clickedChoice = event.target.value;
-    setFieldValues((fieldValues) => {
-      fieldValues = fieldValues || [];
-      // if checked, add to values; if unchecked, remove
-      return checked
-        ? [...fieldValues, clickedChoice]
-        : fieldValues.filter((value) => value !== clickedChoice);
-    });
-  };
-
-  // update form data
-  useEffect(() => {
-    if (fieldValues) {
-      form.setValue(name, fieldValues, { shouldValidate: true });
-    }
-  }, [fieldValues]);
-
-  const errorMessage = form?.formState?.errors?.[name]?.message;
```

Comment:
All of the above has been moved into ChoiceListField - the child component. Same changes in RadioField